### PR TITLE
Use filepath 

### PR DESCRIPTION
**Pull request by sophieyfang**
remove the hardcode path string and use filepath instead.

`go test -mod=mod ./...`  all pass

but for better referencing, the test_data/valid/windows_default_config needs to update the paths to `C:\ProgramData\Google\Cloud Operations\Ops Agent\run\buffers` and decouple the current configenerator_test.go to use different path prefix for windows. Will do in a followup PR

### CODE REVIEW

File: confgenerator/confgenerator.go [hunk 1]
```diff
@@ -249,13 +249,13 @@ func defaultTails(logsDir string, stateDir string) (tails []*conf.Tail) {
 	return []*conf.Tail{
 		{
 			Tag:  "ops-agent-fluent-bit",
-			DB:   fmt.Sprintf("%s/buffers/ops-agent-fluent-bit", stateDir),
-			Path: fmt.Sprintf("%s/logging-module.log", logsDir),
+			DB:   filepath.Join(stateDir, "buffers", "ops-agent-fluent-bit"),
```
Comment (quentinmit) [inline confgenerator/confgenerator.go:252]: It's fine to continue having "buffers/ops-agent-fluent-bit" here.

But, why is this needed at all if we're going to pass it in on the command line when starting fluentbit?

Comment (sophieyfang) [reply confgenerator/confgenerator.go:252]: it will have it in a not pretty syntax.
C:\ProgramData\.../buffers/ops-agent-fluent-bit

Comment (sophieyfang) [reply confgenerator/confgenerator.go:252]: Also for the question why it's needed here, because the configuration of fluent-bit needs to know where to tail the agent's own logs.

File: confgenerator/confgenerator.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ import (
 	"fmt"
 	"sort"
 	"strings"
-
+        "path/filepath"
```
Comment (qingling128) [inline confgenerator/confgenerator.go:22]: `goimports -w .`?

**Review by qingling128 — APPROVED** [review]
LGTM
